router/http/src/main/scala/io/buoyant/router/http/TimestampHeaderFilter.scala

```diff
@@ -0,0 +1,48 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.util.Time
+
+/**
+ * Adds a timestamp header to a message.
+ *
+ * This is intended to be used for the New Relic request queue's
+ * `x-request-start` header, but might be useful for other purposes?
+ */
+object TimestampHeaderFilter {
+
+  val DefaultHeader: String = "X-Request-Start"
+
+  case class Param(header: String) {
```

Comment:
I think this param should contain an `Option[String]` so that this filter can be disabled.  (and it should be disabled by default)